src/python/pants/backend/javascript/package_json.py

```diff
@@ -0,0 +1,631 @@
+    def create(
+        cls,
+        entry_point: str,
+        output_directories: Iterable[str] = (),
+        output_files: Iterable[str] = (),
+        extra_caches: Iterable[str] = (),
+    ) -> NodeBuildScript:
+        return cls(
+            entry_point=entry_point,
+            output_directories=tuple(output_directories),
+            output_files=tuple(output_files),
+            extra_caches=tuple(extra_caches),
+        )
+
+
+class NodePackageScriptsField(SequenceField[NodeBuildScript]):
+    alias = "scripts"
+    expected_element_type = NodeBuildScript
+
+    help = softwrap(
+        """
+        Custom node package manager scripts that should be known
+        and ran as part of relevant goals.
+
+        Maps the package.json#scripts section to a cacheable pants invocation.
+        """
+    )
+    expected_type_description = (
+        '[node_build_script(entry_point="build", output_directories=["./dist/"], ...])'
+    )
+    default = ()
+
+
+class NodePackageVersionField(StringField):
+    alias = "version"
+    help = softwrap(
+        """
+        Version of the Node package, as specified in the package.json.
+
+        This field should not be overridden; use the value from target generation.
+        """
+    )
+    required = True
+    value: str
+
+
+class NodeThirdPartyPackageVersionField(NodePackageVersionField):
+    alias = "version"
+    help = softwrap(
+        """
+        Version of the Node package, as specified in the package.json.
+
+        This field should not be overridden; use the value from target generation.
+        """
+    )
+    required = True
+    value: str
+
+
+class NodePackageNameField(StringField):
+    alias = "package"
+    help = softwrap(
+        """
+        Name of the Node package, as specified in the package.json.
+
+        This field should not be overridden; use the value from target generation.
+        """
+    )
+    required = True
+    value: str
+
+
+class NodeThirdPartyPackageNameField(NodePackageNameField):
+    pass
+
+
+class NodeThirdPartyPackageDependenciesField(Dependencies):
+    pass
+
+
+class NodeThirdPartyPackageTarget(Target):
+    alias = "node_third_party_package"
+
+    help = "A third party node package."
+
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        NodeThirdPartyPackageNameField,
+        NodeThirdPartyPackageVersionField,
+        NodeThirdPartyPackageDependenciesField,
+    )
+
+
+class NodePackageTarget(Target):
+    alias = "node_package"
+
+    help = "A first party node package."
+
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        PackageJsonSourceField,
+        NodePackageNameField,
+        NodePackageVersionField,
+        NodePackageDependenciesField,
+    )
+
+
+class PackageJsonTarget(TargetGenerator):
+    alias = "package_json"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        PackageJsonSourceField,
+        NodePackageScriptsField,
+    )
+    help = softwrap(
+        f"""
```

Comment:
```suggestion
    help = help_text(
        f"""
```

Or to delay processing of f-strings (allowing forward references):


```suggestion
    help = help_text(
        lambda: f"""
```

Reply:
sure!